### PR TITLE
Remove slower defer calls to improve INSERT performance

### PR DESCRIPTION
**Pull request by mugikhan**
This pull request improves the performance of write operations especially INSERTs made outside a transaction context. The query time was reduced from an average time of `~14ms` to `~2ms`.

This was tested using the benchmark test of 1000 INSERTs. The time reduction was noted on both a simulator and a real device.

### CODE REVIEW

File: src/setup-open.ts [hunk 1]
```diff
@@ -119,12 +119,13 @@ export function setupOpen(QuickSQLite: ISQLite) {
                 await hooks?.lockAcquired?.();
                 const res = await callback(context);
 
-                // Ensure that we only resolve after locks are freed
-                _.defer(() => resolve(res));
+                closeContextLock(dbName, id);
+                resolve(res)
               } catch (ex) {
-                _.defer(() => reject(ex));
+                closeContextLock(dbName, id);
```
Comment (stevensJourney) [inline src/setup-open.ts:125]: You should probably remove the call to `closeContextLock` on line 73 to avoid closing the lock twice.

Comment (mugikhan) [reply src/setup-open.ts:125]: Then it won't close the lock of there was an error in the callback? It will skip past that straight to the catch block?

Comment (stevensJourney) [reply src/setup-open.ts:125]: Errors are already caught inside the callback. For the case of an exception (which shouldn't really propagate to that level), the `clsoeContextLock` call could be moved out of the finally block and in to the `catch` block on line 70

File: .changeset/four-meals-grab.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@journeyapps/react-native-quick-sqlite': patch
+---
+
+This pull request improves the performance of write operations especially INSERTs made outside a transaction context.
```
Comment (stevensJourney) [inline .changeset/four-meals-grab.md:5]:
```suggestion
This pull request improves the performance of releasing lock operations. Executing multiple lock operations, such as individual calls to `.execute`, should see a significant performance improvement. 
```

**Review by stevensJourney — APPROVED** [review]

**Pull request merged**